base/applications/rapps/include/appview.h

```diff
@@ -387,8 +387,14 @@ class CApplicationView : public CUiWindow<CWindowImpl<CApplicationView>>
     Create(HWND hwndParent);
     void
     SetRedraw(BOOL bRedraw);
+    enum FocusSelectionMode
+    {
+        FocusSelectUnchanged = 0,
+        FocusSelectAll = MAKELONG(0, 0x7fff),
+        FocusSelectNoneCaretEnd = MAKELONG(0x7fff, 0x7fff),
```

Comment:
Yep this works actually. Note that your 0x7fff could just be `-1` without any problem, thus keeping expectations of the reader with what MSDN mentions regarding EM_SETSEL, and your SendMessage be:
`m_SearchBar->SendMessage(EM_SETSEL, (INT)(INT16)LOWORD(SetSel), (INT)(INT16)HIWORD(SetSel));`
which would be OK (sign-extension)

Reply:
[No, it can't for the FocusSelectNoneCaretEnd case](https://github.com/reactos/reactos/pull/7019#discussion_r1641442250).